pyxem/signals/diffraction2d.py

```diff
@@ -987,48 +978,22 @@ def center_of_mass(
 
         """
         det_shape = self.axes_manager.signal_shape
-        nav_dim = self.axes_manager.navigation_dimension
-        if chunk_calculations is None:
-            chunk_calculations = [16] * nav_dim + list(det_shape)
         if mask is not None:
             x, y, r = mask
-            mask_array = pst._make_circular_mask(x, y, det_shape[0], det_shape[1], r)
-            mask_array = np.invert(mask_array)
-        else:
-            mask_array = None
-        if self._lazy:
-            dask_array = self.data.rechunk(chunk_calculations)
-        else:
-            dask_array = da.from_array(self.data, chunks=chunk_calculations)
-        data = dt._center_of_mass_array(
-            dask_array, threshold_value=threshold, mask_array=mask_array
+            mask = pst._make_circular_mask(x, y, det_shape[0], det_shape[1], r)
+
+        ans = self.map(
+            center_of_mass, threshold=threshold, mask=mask, inplace=inplace, **kwargs
         )
-        if lazy_output:
-            if nav_dim == 2:
-                s_com = LazyDPCSignal2D(data)
-            elif nav_dim == 1:
-                s_com = LazyDPCSignal1D(data)
-            elif nav_dim == 0:
-                s_com = LazyDPCBaseSignal(data).T
+        if inplace:
+            self.T
```

Comment:
If I had to guess I would say this doesn't work?

Reply:
It should work now :) It's tested in `test_center_of_mass_inplace`

Reply:
The reason that it didn't work before was because the transpose function used the `deepcopy_with_new_data` which caused the weird behavior. https://github.com/hyperspy/hyperspy/blob/4039c378f7910f579d5749984534ee2f555bfa8c/hyperspy/signal.py#L6739-L6743